Remove unneeded tests

drvinceknight opened the pull request:
Remove first_play_test and second_play_test with a view to have all
strategies tested via matches.

Have also adjusted the docs.

Closes #1038

marcharper commented on axelrod/tests/strategies/test_alternator.py:
```diff
@@ -21,8 +21,6 @@ class TestAlternator(TestPlayer):
     def test_strategy(self):
-        # Starts by cooperating.
-        self.first_play_test(C)
 
```
Can we eliminate these blank lines? (several more below)

drvinceknight replied:
Yup: lazy by me.

marcharper commented:
Are we sure we're covering all the test cases that second_play_test was covering?

drvinceknight commented:
> Are we sure we're covering all the test cases that second_play_test was covering?

I believe so (I added tests where they were needed and coverage is still at 100%) **but** I'll go through and comment on each one describing what the status is.

drvinceknight commented:
> Are we sure we're covering all the test cases that second_play_test was covering?

I've gone through again and am pretty sure we're good.

Here's a list of all of them (checkboxes in case you want to check):

- [ ] `TestShubik` (axelrod_first): all cases were already covered.
- [ ] `TestCooperator`: all cases were already covered
- [ ] `TestDefector`: all cases were already covered
- [ ] `TestTrickyDefector`: all cases were already covered
- [ ] `TestPSOGambler2_2_2`: all cases were already covered
- [ ] ` TestPSOGambler2_2_2_Noise05`: all cases were already covered
- [ ] ` TestLookerUp`: `(D, D)` was not covered. I have added a test.
- [ ] `TestEvolvedLookerUp2_2_2`: all cases were already covered
- [ ] `TestWinner21`: all cases were already covered
- [ ] `TestWinStayLoseShift`: I added a test against Alternator that covers everything.
- [ ] `TestWinShiftLoseStayTestPlayer`: I added a test against Alternator that covers everything.
- [ ] `TestNegation`: I added 4 versus tests that cover everything.
- [ ] `TestTitForTat`: all cases were covered.
- [ ] `TestTitFor2Tats`: I added one extra turn to a test that was already there (it now covers everything).
- [ ] `TestTwoTitsForTat`: I added versus tests against cooperator and defector to cover `(C, C)` and `(D, D)`.
- [ ] `TestBully`: all cases were covered.
- [ ] `TestSneakyTitForTat`: all cases were covered.
- [ ] `TestSuspiciousTitForTat`: I added versus tests against cooperator and defector to cover `(C, C)` and `(D, D)`.
- [ ] `TestAntiTitForTat`: all cases were covered.
- [ ] `TestHardTitForTat`: I added versus tests against cooperator and defector to cover `(C, C)` and `(D, D)`.
- [ ] `TestOmegaTFT]: all cases were covered.
- [ ] `TestAdaptiveTitForTat`: I added a versus test against Alternator
- [ ] `TestSpitefulTitForTat`: I added one extra turn to cover `(D, C)`
- [ ] `TestAlexei`: all cases were covered
- [ ] `TestEugineNier`: all cases were covered

marcharper commented:
Looks good to me -- I see now that some of the single tests are catching all four possible contexts.

Likely there will be merge conflicts with #1037

drvinceknight commented:
> Likely there will be merge conflicts with #1037

They're not too bad, I've got a rebased branch locally. Once one of these goes in I can patch the other.

meatballs commented:
I'll merge this one first. Still waiting for Travis in #1037
